### PR TITLE
📖 Move clusterctl to book

### PR DESCRIPTION
**Pull request by kikisdeliveryservice**
Move clusterctl to book   
    - moved docs/how-to-use-clusterctl.md to docs/book/src/tooling/clusterctl.md
    - deleted docs/book/src/tooling/clusterctl.md
    - remove ref to clusterctl.md from /hack/verify-doctoc.sh

Fixes: #1372

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
hmm in the original: https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/how-to-use-clusterctl.md#what-is-clusterctl  the link in provider implementations just goes to the repo itself?

**Comment by kikisdeliveryservice** [comment]
@vincepri 2 questions (since I'm new here):

For this PR should I delete the existing how-to-user-clusterctl.md doc as well?

Should i update: https://github.com/kubernetes-sigs/cluster-api/blob/1f6bf657568e3806ed89725d06222590afba11b5/hack/verify-doctoc.sh#L26 to include this file now?

**Comment by vincepri** [comment]
@kikisdeliveryservice It should be ok to remove the file and the reference as well, we should remove the TOC from the file given that the book doesn't require it

**Comment by kikisdeliveryservice** [comment]
> @kikisdeliveryservice It should be ok to remove the file and the reference as well, we should remove the TOC from the file given that the book doesn't require it

Thanks! Done.

File: docs/book/src/tooling/clusterctl.md [hunk 1]
```diff
@@ -1 +1,85 @@
...
+For more details about all the supported options run:
+
+```
+clusterctl create cluster --help
+```
+
+After generating the YAML files run the following command:
+
+```
+clusterctl create cluster --provider <PROVIDER> --bootstrap-type <BOOTSTRAP CLUSTER TYPE> -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
```
Comment (detiber) [inline docs/book/src/tooling/clusterctl.md]: I think the `--provider` field no longer exists with v1alpha2, @vincepri can you confirm?

File: docs/book/src/tooling/clusterctl.md [hunk 2]
```diff
@@ -1 +1,85 @@
...
+
+# Amazon AWS
+clusterctl create cluster --provider aws --bootstrap-type kind -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
+```
+
+**What happens when we run the command?**  
+After running the command first it creates a local cluster. If `kind` was passed as the `--bootstrap-type`
+it creates a local [kind](https://kind.sigs.k8s.io/) cluster. This cluster is generally referred to as the *bootstrap cluster*.  
+On this kind Kubernetes cluster the `provider-components.yaml` file is applied. This step loads the CRDs into
+the cluster. It also creates two [StatefulSet](https://Kubernetes.io/docs/concepts/workloads/controllers/statefulset/)
```
Comment (detiber) [inline docs/book/src/tooling/clusterctl.md]: s/StatefulSet/Deployment/ here, with v1alpha2 and using the updated kubebuilder v2 scaffolding the old StatefulSets should now be Deployments.

File: docs/book/src/tooling/clusterctl.md [hunk 3]
```diff
@@ -1 +1,85 @@
...
+pods that run the cluster api controller and the provider specific controller. These pods register the custom
```
Comment (detiber) [inline docs/book/src/tooling/clusterctl.md]:
```suggestion
pods that run the cluster api controller manager and the provider specific controller manager. These pods register the custom
```

File: docs/book/src/tooling/clusterctl.md [hunk 4]
```diff
@@ -1 +1,85 @@
...
+controllers that manage the newly defined resources (`Cluster`, `Machine`, `MachineSet`, etc.).
```
Comment (detiber) [inline docs/book/src/tooling/clusterctl.md]:
```suggestion
controllers that manage the newly defined resources (`Cluster`, `Machine`, `MachineSet`, `MachineDeployment`, as well as provider-specific resources).  
```

File: docs/book/src/tooling/clusterctl.md [hunk 5]
```diff
@@ -1 +1,85 @@
...
+
+Next, `clusterctl` applies the `cluster.yaml` and `machines.yaml` to the local kind Kubernetes cluster. This
+step creates a Kubernetes cluster with only a control-plane (as defined in `machines.yaml`) on the specified
+provider. This newly created cluster is generally referred to as the *management cluster* or *pivot cluster*
+or *initial target cluster*. The management cluster is responsible for creating and maintaining the workload cluster.  
+  
+Lastly, `clusterctl` moves all the CRDs and the custom controllers from the bootstrap cluster to the
+management cluster and deletes the locally created bootstrap cluster. This step is referred to as the *pivot*.
+
+### Creating a workload cluster using the management cluster
```
Comment (detiber) [inline docs/book/src/tooling/clusterctl.md]: We should likely remove this section, I'm not sure that it works as expected.

File: docs/book/src/tooling/clusterctl.md (hunk 1 above)
Comment (vincepri) [reply docs/book/src/tooling/clusterctl.md]: That's correct, good catch!

**Comment by kikisdeliveryservice** [comment]
Thanks @detiber I'll get those suggestions in later today. :smiley_cat:

File: docs/book/src/tooling/clusterctl.md (hunk 1 above)
Comment (kikisdeliveryservice) [reply docs/book/src/tooling/clusterctl.md]: done.

File: docs/book/src/tooling/clusterctl.md (hunk 2 above)
Comment (kikisdeliveryservice) [reply docs/book/src/tooling/clusterctl.md]: done

File: docs/book/src/tooling/clusterctl.md (hunk 3 above)
Comment (kikisdeliveryservice) [reply docs/book/src/tooling/clusterctl.md]: done

File: docs/book/src/tooling/clusterctl.md (hunk 4 above)
Comment (kikisdeliveryservice) [reply docs/book/src/tooling/clusterctl.md]: done

File: docs/book/src/tooling/clusterctl.md (hunk 5 above)
Comment (kikisdeliveryservice) [reply docs/book/src/tooling/clusterctl.md]: done

**Comment by kikisdeliveryservice** [comment]
Updated as per comments, @detiber and @vincepri PTAL.

File: docs/book/src/tooling/clusterctl.md [hunk 6]
```diff
@@ -1 +1,69 @@
...
+```
+
+Example usage:
+
+```
+# VMware vSphere
+clusterctl create cluster --provider vsphere --bootstrap-type kind -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
+
+# Amazon AWS
+clusterctl create cluster --provider aws --bootstrap-type kind -c cluster.yaml -m machines.yaml -p provider-components.yaml --addon-components addons.yaml
```
Comment (vincepri) [inline docs/book/src/tooling/clusterctl.md]: We can remove both of the above `--provider` flags, they have been deprecated in v1alpha2

Comment (vincepri) [reply docs/book/src/tooling/clusterctl.md]: Follow-up, we can also remove the two examples and just keep a generic one

File: docs/book/src/tooling/clusterctl.md [hunk 7]
```diff
@@ -1 +1,69 @@
-# Clusterctl
+# Using `clusterctl` to create a cluster from scratch
```
Comment (vincepri) [inline docs/book/src/tooling/clusterctl.md]: Above this line, we should add the following deprecation notice:

```html
<aside class="note warning">

<h1>Deprecation Notice</h1>

This utility has been deprecated in v1alpha2 and it'll be removed in a future version in favor. 

</aside>
```

File: docs/book/src/tooling/clusterctl.md (hunk 6 above)
Comment (kikisdeliveryservice) [reply docs/book/src/tooling/clusterctl.md]: gotcha. done.

File: docs/book/src/tooling/clusterctl.md (hunk 7 above)
Comment (kikisdeliveryservice) [reply docs/book/src/tooling/clusterctl.md]: done.

**Pull request merged**